Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp

```diff
@@ -182,6 +191,33 @@ void WPEQtView::didUpdateScene()
     wpe_view_qtquick_did_update_scene(WPE_VIEW_QTQUICK(wpeView));
 }
 
+void WPEQtView::updateHttpAcceptLanguage()
+{
+    if (!m_webView)
+        return;
+
+    auto* wpeContext = webkit_web_view_get_context(m_webView.get());
```

Comment:
What this is doing is fetching the _existing_ `WebKitWebContext` from the web view and modifying the existing instance. A new instance is never created, which means this is doing changes to the default web context created internally by WebKit when one is not specified manually for a web view&mdash;and not only for the web views which have a certain `WPEQtProfile` assigned!

I would expect that `WPEQtProfile` wraps a `WebKitWebContext` (and, in the future, an associated `WebKitWebsiteDataManager`), which is what gets modified. Then when creating a `WPEQtView` we pass the `WPEQtProfile` that the view has to use, and finally when the underlying `WebKitWebView` is instantiated, the `WebKitWebContext` wrapped by the `WPEQtProfile` gets passed as the `web-context` construct-property.

Looking at the [QWebEnginePage documentation](https://doc.qt.io/qt-6/qwebenginepage.html), the `QWebEngineProfile` needs to be passed at construction time, and there is no way to change it afterwards, so trying to do the same as the QtWebEngine API actually fits well with the intended usage of the WPE GLib API.

Reply:
As discussed on Matrix, the `profile` property is exposed by `WebEngineView` (the QML type), https://doc.qt.io/qt-6/qml-qtwebengine-webengineview.html#profile-prop.

The `WebEngineView` is first constructed, and the `profile` property is then set using a `setProperty` call. Apparently this is how QWebEngine works internally.